src/main/java/appeng/parts/reporting/PartThroughputMonitor.java

```diff
@@ -0,0 +1,205 @@
+
+    private static final String[] TIME_UNIT = { "/t", "/s", "/m", "/h" };
+    private static final float[] NUMBER_MULTIPLIER = { 1, 20, 1_200, 72_000 };
+
+    private double itemNumsChange;
+    private int timeMode;
+    private long lastStackSize;
+
+    @Reflected
+    public PartThroughputMonitor(final ItemStack is) {
+        super(is);
+        this.itemNumsChange = 0;
+        this.timeMode = 0;
+        this.lastStackSize = -1;
+    }
+
+    @Override
+    public CableBusTextures getFrontBright() {
+        return FRONT_BRIGHT_ICON;
+    }
+
+    @Override
+    public CableBusTextures getFrontColored() {
+        return this.isLocked() ? FRONT_COLORED_ICON_LOCKED : FRONT_COLORED_ICON;
+    }
+
+    @Override
+    public CableBusTextures getFrontDark() {
+        return FRONT_DARK_ICON;
+    }
+
+    @Override
+    public void readFromNBT(final NBTTagCompound data) {
+        super.readFromNBT(data);
+        this.timeMode = data.getInteger("timeMode");
+    }
+
+    @Override
+    public void writeToNBT(final NBTTagCompound data) {
+        super.writeToNBT(data);
+        data.setInteger("timeMode", this.timeMode);
+    }
+
+    @Override
+    public void writeToStream(final ByteBuf data) throws IOException {
+        super.writeToStream(data);
+        data.writeInt(this.timeMode);
+        data.writeDouble(this.itemNumsChange);
+    }
+
+    @Override
+    public boolean readFromStream(final ByteBuf data) throws IOException {
+        boolean needRedraw = super.readFromStream(data);
+        this.timeMode = data.readInt();
+        this.itemNumsChange = data.readDouble();
+        return needRedraw;
+    }
+
+    @Override
+    public boolean onPartShiftActivate(final EntityPlayer player, final Vec3 pos) {
+        if (Platform.isClient()) {
+            return true;
+        }
+
+        if (!this.getProxy().isActive()) {
+            return false;
+        }
+
+        if (!Platform.hasPermissions(this.getLocation(), player)) {
+            return false;
+        }
+
+        this.timeMode = (this.timeMode + TIME_UNIT.length - 1) % TIME_UNIT.length;
+        this.host.markForUpdate();
+
+        return true;
+    }
+
+    @Override
+    public boolean onPartActivate(final EntityPlayer player, final Vec3 pos) {
+        if (Platform.isClient()) {
+            return true;
+        }
+
+        if (!this.getProxy().isActive()) {
+            return false;
+        }
+
+        if (!Platform.hasPermissions(this.getLocation(), player)) {
+            return false;
+        }
+
+        final TileEntity te = this.getTile();
+        final ItemStack eq = player.getCurrentEquippedItem();
+
+        if (!Platform.isWrench(player, eq, te.xCoord, te.yCoord, te.zCoord) && this.isLocked()) {
+            this.timeMode = (this.timeMode + 1) % TIME_UNIT.length;
+            this.host.markForUpdate();
+            return true;
+        } else {
+            return super.onPartActivate(player, pos);
+        }
+    }
```

Comment:
I don't think it's necessary to override the regular right-click behavior. Shift-clicking to change the mode is sufficient.